cmd/lakectl/cmd/branch_tools_test.go

```diff
@@ -0,0 +1,117 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/treeverse/lakefs/pkg/metastore"
+	"github.com/treeverse/lakefs/pkg/metastore/mock"
+)
+
+const (
+	dbName     = "source"
+	repoName   = "repo"
+	branchName = "branch"
+)
+
+var (
+	validPath   = fmt.Sprintf("s3://%s/%s/path/to/schema", repoName, branchName)
+	invalidPath = fmt.Sprintf("s3/%s/%s/", repoName, branchName)
+)
+
+type extractRepoAndBranchFromDBNameTestCase struct {
+	name string
+	args struct {
+		dbName string
+	}
+	repositoryName string
+	branchName     string
+	errType        error
+	validPath      bool
+}
+
+func TestExtractRepoAndBranchFromDBName(t *testing.T) {
+	type args struct {
+		dbName string
+	}
+	tests := []extractRepoAndBranchFromDBNameTestCase{
+		{
+			name: "Sunny day flow",
+			args: args{
+				dbName: dbName,
+			},
+			repositoryName: repoName,
+			branchName:     branchName,
+			errType:        nil,
+			validPath:      true,
+		},
+		{
+			name: "Client with no database",
+			args: args{
+				dbName: "db name that doesn't exist",
+			},
+			repositoryName: "",
+			branchName:     "",
+			errType:        MissingDBError{},
+			validPath:      true,
+		},
+		{
+			name: "Failed getting repo and branch from uri",
+			args: args{
+				dbName: dbName,
+			},
+			repositoryName: "",
+			branchName:     "",
+			errType:        ExtractSourceBranchError{},
+			validPath:      false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var schemaPath string
+			if tt.validPath {
+				schemaPath = validPath
+			} else {
+				schemaPath = invalidPath
+			}
```

Comment:
Why not passing the correct constant in the test data? It will also open the case where we like to pass something else like empty string.